apply/apply.go

```diff
@@ -204,7 +239,19 @@ func applyEnvs(
 			}
 
 			if componentPlan.ModuleSource != nil {
-				e := applyModuleInvocation(fs, path, *componentPlan.ModuleSource, componentPlan.ModuleName, templates.Templates.ModuleInvocation, commonBox)
+				downloader := util.MakeDownloader(*componentPlan.ModuleSource)
+				if oauth {
+					type HTTPAuth struct {
+						GithubToken string `required:"true"`
+					}
+					var httpAuth HTTPAuth
+					err := envconfig.Process("fogg", &httpAuth)
+					if err != nil {
+						return errs.WrapUser(err, "unable to get env FOGG_GITHUBTOKEN with the --oauth flag")
+					}
```

Comment:
should all of this logic instead be embedded as part of the downloader itself? otherwise users of this code need to be making these decisions each time and it will be out of sync eventually